### PR TITLE
Add read pool

### PR DESCRIPTION
**Pull request by breeswish**
This PR extracted the read pool from https://github.com/breeswish/tikv/pull/1
Changes in `kv.rs` are not included in this PR.

Some part of this PR has been extracted into another PR: https://github.com/pingcap/tikv/pull/2685 

Currently read pool does not support statistics.

@zhangjinpeng1987 @siddontang @hicqu PTAL

### CODE REVIEW

File: tests/config/test-custom.toml [hunk 1]
```diff
@@ -16,6 +16,12 @@ end-point-stack-size = "12MB"
 end-point-recursion-limit = 100
 end-point-batch-row-limit = 64
 snap-max-write-bytes-per-sec = "10MB"
+readpool-read-critical-concurrency = 2
```
Comment (siddontang) [inline tests/config/test-custom.toml]: do we have the critical priority command now?

File: tests/config/test-custom.toml [hunk 2]
```diff
@@ -16,6 +16,12 @@ end-point-stack-size = "12MB"
...
+readpool-read-high-concurrency = 2
```
Comment (siddontang) [inline tests/config/test-custom.toml]: I prefer using an independent section for read pool.

File: src/readpool/errors.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::error;
+use storage;
+
+quick_error!{
+    #[derive(Debug)]
+    pub enum Error {
+        SchedulerBusy(task_detail: String) {
```
Comment (siddontang) [inline src/readpool/errors.rs]: why should readpool know the scheduler error?

File: tests/config/test-custom.toml (hunk 1 above)
Comment (breeswish) [reply tests/config/test-custom.toml]: Critical priority is planned for operations like schema queries.

Comment (siddontang) [reply tests/config/test-custom.toml]: if we don't use it now, remove it.

**Pull request closed without merging**